app/app.go

```diff
@@ -135,8 +137,26 @@ func MakeCodec() *codec.Codec {
 	return cdc.Seal()
 }
 
-// https://github.com/cosmos/sdk-tutorials/blob/c6754a1e313eb1ed973c5c91dcc606f2fd288811/app.go#L73
+// chainAnteChains point of this function is to sequentially run multiple chains
+// like it would be one single chain. It becomes useful in case app implements own
+// AnteHandlers and app wants run them with all default handlers from the sdk
+// Purpose of the function came from an issue that sdk.AnteHandler does not have next parameter
+// and default chain can be accessed only from ante.NewAnteHandler which returns sdk.AnteHandler.
+// It eliminates possibility for apps to chain own Ante
+func chainAnteChains(chains ...sdk.AnteHandler) sdk.AnteHandler {
```

Comment:
Can you put this in a different file?

It consumes handlers and produces a handler, I still think the name is a bit off, but that's okay I guess.

Reply:
`tools.go` in app ?

Reply:
sure sounds good.  or `utils.go`.  thank you.